Dockerfile

```diff
@@ -30,13 +30,15 @@ WORKDIR /iroha
 COPY . .
 RUN cargo chef prepare --recipe-path recipe.json
 
-ARG PROFILE=
+ARG PROFILE=""
 FROM cargo-chef as cacher
+ARG PROFILE
 WORKDIR /iroha
 COPY --from=planner /iroha/recipe.json recipe.json
 RUN cargo chef cook $PROFILE --recipe-path recipe.json
 
 FROM rust-base as builder
+ARG PROFILE
```

Comment:
This should be the same argument. And they are specified before FROM

Reply:
Any FROM consume provided ARG so they are not available after build stage later. [Docs](https://docs.docker.com/engine/reference/builder/#understand-how-arg-and-from-interact)

Reply:
So it can be putted before FROM or at the begininning of the file. It should work this way as global argument

Reply:
That's not correct. Please read the [docs](https://docs.docker.com/engine/reference/builder/#understand-how-arg-and-from-interact). Even global args need to be redeclared if it was consumed by intermediate build stage earlier